enumerables.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x])
+      x += 1
+    end
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    x = 0
+    selfitem = self
+    while x < selfitem.size
+      yield(selfitem[x], x)
+      x += 1
+    end
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    array = []
+    selfitem = self
+    selfitem.my_each { |x| array << x if yield(x) }
+    array
+  end
+
+  def my_all?
+    selfitem = self
+    if !block_given?
+      selfitem.my_each do |x|
+        return true unless x != false
+      end
+    elsif block_given?
+      selfitem.my_each { |x| return false unless yield(x) }
+    else
+      selfitem.my_each { |x| return false unless x }
+    end
+    true
+  end
+
+  def my_any?
+    selfitem = self
+    if !block_given?
+      selfitem.my_each do |x|
+        return true if x == true
+      end
+    elsif block_given?
+      selfitem.my_each { |x| return true if yield(x) }
+    else
+      selfitem.my_each { |x| return true if yield(x) }
+    end
+    false
+  end
```

Comment:
**[REQUIRED]**
* When a class is passed as an argument returns true if at least one of the collection is a member of such class.
```
Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a Regex is passed as an argument returns false if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```
* When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
     ArgumentError:
       wrong number of arguments (given 1, expected 0)
```